Source/Core/VideoCommon/AVIDump.cpp

```diff
@@ -94,47 +95,56 @@ bool AVIDump::Start(int w, int h)
 
 bool AVIDump::CreateVideoFile()
 {
-  AVCodec* codec = nullptr;
-
-  s_format_context = avformat_alloc_context();
-  std::stringstream s_file_index_str;
-  s_file_index_str << s_file_index;
-  snprintf(s_format_context->filename, sizeof(s_format_context->filename), "%s",
-           (File::GetUserPath(D_DUMPFRAMES_IDX) + "framedump" + s_file_index_str.str() + ".avi")
-               .c_str());
-  File::CreateFullPath(s_format_context->filename);
-
-  // Ask to delete file
-  if (File::Exists(s_format_context->filename))
+  const char* s_format = g_Config.sDumpFormat.c_str();
+  
+  std::string s_dump_path = g_Config.sDumpURL;
+  
+  if (s_dump_path.empty())
   {
-    if (SConfig::GetInstance().m_DumpFramesSilent ||
-        AskYesNoT("Delete the existing file '%s'?", s_format_context->filename))
-    {
-      File::Delete(s_format_context->filename);
-    }
-    else
+    std::stringstream file_ss;
+    file_ss << File::GetUserPath(D_DUMPFRAMES_IDX)
+                     << "framedump" << s_file_index
+                     << "." << s_format;
+    s_dump_path = file_ss.str();
+    
+    File::CreateFullPath(s_dump_path);
+
+    // Ask to delete file
+    if (File::Exists(s_dump_path))
     {
-      // Stop and cancel dumping the video
-      return false;
+      if (SConfig::GetInstance().m_DumpFramesSilent ||
+          AskYesNoT("Delete the existing file '%s'?", s_dump_path.c_str()))
+      {
+        File::Delete(s_dump_path);
+      }
+      else
+      {
+        // Stop and cancel dumping the video
+        return false;
+      }
     }
   }
 
-  if (!(s_format_context->oformat = av_guess_format("avi", nullptr, nullptr)))
+  AVOutputFormat* output_format = av_guess_format(s_format, s_dump_path.c_str(), nullptr);
+  if (!output_format)
   {
+    WARN_LOG(VIDEO, "Invalid format %s", s_format);
     return false;
   }
+  avformat_alloc_output_context2(&s_format_context, output_format, nullptr, s_dump_path.c_str());
```

Comment:
This should probably have its return value checked for the case allocation fails.